go/libkb/merkle_client.go

```diff
@@ -305,68 +524,113 @@ func (mc *MerkleClient) LookupPath(ctx context.Context, q HTTPArgs, sigHints *Si
 	})
 
 	if err != nil {
-		return
+		return nil, nil, nil, err
 	}
 	switch res.AppStatus.Code {
 	case SCNotFound:
 		err = NotFoundError{}
-		return
+		return nil, nil, nil, err
 	case SCDeleted:
 		err = DeletedError{}
-		return
+		return nil, nil, nil, err
 	}
 
 	if sigHints != nil {
 		if err = sigHints.RefreshWith(ctx, res.Body.AtKey("sigs")); err != nil {
-			return
+			return nil, nil, nil, err
 		}
 	}
 
-	root, err := NewMerkleRootFromJSON(res.Body.AtKey("root"), mc.G())
+	var ret *VerificationPath
+	ret, err = mc.readPathFromAPIRes(ctx, res)
 	if err != nil {
-		return
+		return nil, nil, nil, err
+	}
+	ss, err = mc.readSkipSequenceFromAPIRes(ctx, res, ret.root, lastRoot)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	return ret, ss, res, err
+}
+
+func (mc *MerkleClient) readSkipSequenceFromAPIRes(ctx context.Context, res *APIRes, thisRoot *MerkleRoot, lastRoot *MerkleRoot) (ret SkipSequence, err error) {
+	defer mc.G().CTrace(ctx, "MerkleClient#readSkipSequenceFromAPIRes", func() error { return err })()
+	if lastRoot == nil {
+		mc.G().Log.CDebugf(ctx, "| lastRoot==nil")
+		return nil, nil
+	}
+	if !thisRoot.HasSkips() {
+		mc.G().Log.CDebugf(ctx, "| thisRoot has no skips")
+		return nil, nil
+	}
+	skips := res.Body.AtKey("skips")
+
+	if skips.IsNil() {
+		mc.G().Log.CDebugf(ctx, "| skip list from API server is nil")
+		return nil, nil
+	}
+
+	var v []string
+	if err = skips.UnmarshalAgain(&v); err != nil {
+		mc.G().Log.CDebugf(ctx, "| failed to unmarshal skip list as a list of strings")
+		return nil, err
+	}
+
+	ret = SkipSequence{thisRoot.payload}
+
+	for _, s := range v {
+		var p MerkleRootPayload
+		if p, err = NewMerkleRootPayloadFromJSONString(s); err != nil {
+			return nil, err
+		}
+		ret = append(ret, p)
+	}
+	ret = append(ret, lastRoot.payload)
```

Comment:
Ah :p I was _just_ reading over this, after writing that comment above, and I _still_ had to come back and remind myself that `thisRoot` makes it in this list too up on line 579. Maybe a doc comment on the function itself here?